src/rdpc/analysesProcessor.ts

```diff
@@ -231,10 +234,16 @@ export const getAllMergedDonor = async ({
 export const countSpecimenType = (map: StringMap<DonorData>): DonorInfoMap => {
   const result: DonorInfoMap = {};
   Object.entries(map).forEach(([donorId, donorData]) => {
-    for (const specimen of donorData.specimen) {
-      if (
-        specimen.tumourNormalDesignation === TumourNormalDesignationValue.Normal
-      ) {
+    const normalSpecimen = donorData.specimen.filter(
+      (sp) => sp.tumourNormalDesignation === TumourNormalDesignationValue.Normal
+    );
+
+    const tumourSpecimen = donorData.specimen.filter(
+      (sp) => sp.tumourNormalDesignation === TumourNormalDesignationValue.Tumour
+    );
+
+    for (const normal of normalSpecimen) {
+      if (normal.samples[0].sampleType.includes(DNA_SAMPLE_TYPE_KEYWORD)) {
```

Comment:
Is there any risk that `normal.samples[0]` will be null? Should this be `normal.samples[0]?.sampleType.includes(DNA_SAMPLE_TYPE_KEYWORD)`.